ocaml/lambda/switch.ml

```diff
@@ -41,20 +42,27 @@ module CtxStore(A:CtxStored) = struct
   module AMap =
     Map.Make(struct type t = A.key let compare = A.compare_key end)
 
+  type act =
+    { shared : bool;
+      act : A.t;
+    }
+
+  module Int = Numbers.Int
+
   type intern =
     { mutable map : (bool * int)  AMap.t ;
       mutable next : int ;
-      mutable acts : (bool * A.t) list; }
+      acts : act Int.Tbl.t; }
```

Comment:
I would have thought it was a little simpler to use:
```ocaml
type act =
    { mutable shared : bool;
      mutable act : A.t;
    }

type intern =
    { mutable map : act  AMap.t ;
      mutable next : int ;
      mutable acts : act list; }
```

Then you can just look up the key in the map and edit the action and sharedness, and at the end you can just read the results straight off the list.